pkg/driver/node_server.go

```diff
@@ -1621,6 +1622,20 @@ func (driver *Driver) NodeGetInfo(ctx context.Context, req *csi.NodeGetInfoReque
 		nodeMaxVolumesLimit = defaultMaxVolPerNode
 	}
 
+	// Create a anonymous wrapper function over nodeGetInfo. This function is called
+	// periodically.
+	getNodeInfoFunc := func() {
+		nodeID, err = driver.nodeGetInfo()
+		if err != nil {
+			log.Errorf("Failed to update %s nodeInfo. Error: %s", nodeID, err.Error())
+		}
+		return
+	}
+	// Initialize polling with anonymous wrapper function.
```

Comment:
I think GetNodeInfo might not be the right place to add this task, as there can be possibility for kubelet to call this function multiple times adding duplicate tasks. Can we add this as part of NodeDriver init itself?

Reply:
Good catch Shiva. thanks. 
CO may call this GetNodeInfo rpc more than once although I have not noticed during testing. As mostly CO invokes NodeGetCapabilities rpc multiple times. I will make the code changes .